pkg/controller/dgraph/models/query/login.go

```diff
@@ -54,7 +54,7 @@ func hashAndUpdatePassword(login *dgraph.Login, newPassword string) error {
 // GetHashedPassword ...
 func GetHashedPassword(username string) (dgraph.Login, error) {
 	q := `query {
-		login(func: has(isLogin)) {
+		login(func: has(isLogin)) @filter(eq(username, "` + username + `")) {
```

Comment:
We should make sure `username` doesn't contain special string characters such as _The grave accent (`)_ or _quotes_ etc.

This is needed because, if the username is set as `randomUsername) OR eq(1, 1)` which will make the Dgraph filter as `eq(username, randomUsername) OR eq(1, 1)` which will return true for any `randomUsername`

Reply:
Also, I wrongly name the `GetHashedPassoword` before. If possible change it as it returns the login (username and saltedHashPassword)